### PR TITLE
GAWB-3122 Rawls /status calls Cromwell /status

### PR DESCRIPTION
**Pull request by jmthibault79**
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [x] **Submitter**: Make sure Swagger is updated if API changes
  - [x] **...and Orchestration's Swagger too!**
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/MockShardedExecutionServiceCluster.scala [hunk 1]
```diff
@@ -3,7 +3,7 @@ package org.broadinstitute.dsde.rawls.dataaccess
 /**
   * Created by davidan on 6/16/16.
   */
-class MockShardedExecutionServiceCluster(readMembers: Map[ExecutionServiceId, ExecutionServiceDAO], submitMembers: Map[ExecutionServiceId, ExecutionServiceDAO], dataSource: SlickDataSource)
+class MockShardedExecutionServiceCluster(val readMembers: Map[ExecutionServiceId, ExecutionServiceDAO], submitMembers: Map[ExecutionServiceId, ExecutionServiceDAO], dataSource: SlickDataSource)
```
Comment (jmthibault79) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/dataaccess/MockShardedExecutionServiceCluster.scala:6]: so `readMembers` can be accessed by `ApiServiceSpec`

**Review by helgridly — APPROVED** [review]
I am inexpert but this seems good. Maybe add a Cromwellian too

**Comment by jmthibault79** [comment]
jenkins retest

File: core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpExecutionServiceDAO.scala [hunk 2]
```diff
@@ -81,6 +83,11 @@ class HttpExecutionServiceDAO( executionServiceURL: String, submissionTimeout: F
     retry(when500) { () => pipelineNoAuth[ExecutionServiceVersion] apply Get(url) }
   }
 
+  override def getStatus(): Future[Map[String, SubsystemStatus]] = {
+    val url = executionServiceURL + s"/engine/v1/status"
+    retry(when500) { () => pipelineNoAuth[Map[String, SubsystemStatus]] apply Get(url) }
```
Comment (rtitle) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpExecutionServiceDAO.scala]: Is retrying on 500 errors appropriate here? Maybe we should just interpret a 500 response as a non-ok status.

File: core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/HealthMonitor.scala [hunk 3]
```diff
@@ -132,11 +138,21 @@ class HealthMonitor private (val slickDataSource: SlickDataSource, val googleSer
...
   private def checkCromwell: Future[SubsystemStatus] = {
-    logger.debug("Checking Cromwell...")
-    Future.successful(OkStatus)
+    logger.debug("Checking Cromwell(s)...")
+
+    def annotateSubsystem(serviceId: ExecutionServiceId, execSubsystem: String, subsystemStatus: SubsystemStatus): SubsystemStatus = {
+      subsystemStatus.copy(messages = subsystemStatus.messages.map { msgList => msgList.map { msg => s"${serviceId.id}-$execSubsystem: $msg" } })
+    }
+
+    // Note: calls to `foldMap` depend on SubsystemStatusMonoid
```
Comment (rtitle) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/HealthMonitor.scala:150]: Nice way to fold a `List[Future[List[SubsystemStatus]]]` to a `Future[SubsystemStatus]`

File: core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpExecutionServiceDAO.scala (hunk 2 above)
Comment (dvoet) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpExecutionServiceDAO.scala]: I say leave the retries, at least for now, so we can be resilient towards short lived glitches

Comment (rtitle) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpExecutionServiceDAO.scala]: Agora status doesn't retry: https://github.com/broadinstitute/rawls/blob/develop/core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpMethodRepoDAO.scala#L85

**Review by rtitle — CHANGES_REQUESTED** [review]
Looks good, I just think we should remove that retry

File: core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpExecutionServiceDAO.scala (hunk 2 above)
Comment (jmthibault79) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpExecutionServiceDAO.scala]: that's a good point - 500 is actually how it reports non-ok

File: core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/HealthMonitor.scala (hunk 3 above)
Comment (dvoet) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/HealthMonitor.scala:150]: my mother was always bad at folding maps

**Review by dvoet — APPROVED** [review]

**Comment by jmthibault79** [comment]
removed retry

**Comment by jmthibault79** [comment]
jenkins retest

**Review by rtitle — APPROVED** [review]

**Pull request merged**